### PR TITLE
test for user

### PR DESCRIPTION
**Pull request by petermason**
# Description

_Description of the feature or fix with link to the issue if existing_

FIXES:

from homescreen, log out
land on the 3 flags page
PRESS THE BACK BUTTON
Relaunch the app, either directly to the homescreen (on a Samsung A50) or back to 3 flags on my OnePlus 6T.
if relaunched to 3 flags, press the back button.
if app exits, relaunch it
=> HOME SCREEN!

## On what platform have you tested the change?

- [X] Android - Emulator
- [ ] Android - Physical device
- [ ] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/core/user/UserService.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export default class UserService extends ApiClientBase implements IUserService {
   public async logout() {
     this.hasUser = false;
     await this.deleteLocalUserData();
-    NavigatorService.navigate('CountrySelect');
```
Comment (selvach) [inline src/core/user/UserService.ts:78]: Did you mean to leave this commented out?

Comment (petermason) [reply src/core/user/UserService.ts:78]: i actually meant to remove it! It gets called in the Logout.hooks.ts file

**Review by selvach — APPROVED** [review]

File: src/core/user/UserService.ts [hunk 2]
```diff
@@ -75,7 +75,6 @@ export default class UserService extends ApiClientBase implements IUserService {
   public async logout() {
     this.hasUser = false;
```
Comment (andyguest) [inline src/core/user/UserService.ts:76]: Why don't we just remove this.patientData here too?   And any other things that may now be invalid.

Or if we can't do that, see my other question.

File: src/features/AppCoordinator.ts [hunk 3]
```diff
@@ -68,6 +68,8 @@ export class AppCoordinator extends Coordinator implements SelectProfile, Editab
 
   shouldShowCountryPicker: boolean = false;
 
+  thisUser: UserResponse | null;
+
   screenFlow: Partial<ScreenFlow> = {
     Splash: () => {
       if (this.patientData && this.shouldShowCountryPicker) {
```
Comment (andyguest) [inline src/features/AppCoordinator.ts:75]: It's not safe for us to look at this.patientData before validating we have this.thisUser.  I think this sequence of code arms should be reordered / protected better.  

Is there potential for this kind of bug to exhibit when we remove and reinstall?  Did we make the necessary changes to remove ALL local data on uninstall of our app?

**Review by andyguest — COMMENTED** [review]
I've attached a couple of questions to this.  They don't need to impact this PR, but might be food for further thought.

File: src/features/AppCoordinator.ts (hunk 3 above)
Comment (selvach) [reply src/features/AppCoordinator.ts:75]: AsyncStorage is wiped on uninstall on iOS and Android.

**Pull request merged**